autode/utils.py

```diff
@@ -21,10 +23,51 @@
     CouldNotGetProperty,
 )
 
-try:
-    mp.set_start_method("fork")
-except RuntimeError:
-    logger.warning("Multiprocessing context has already been defined")
+
+def copy_into_current_config(
+    parent_config: "autode.config._ConfigClass",
+) -> None:
+    """
+    Copies an instance of Config into current process. Required to set the
+    process pool workers' Config to the same state as the parent, when
+    not forking the interpreter. To be only run on initializing workers
+
+    Args:
+        parent_config: Parent config instance that will be copied into
+         present process
+    """
+    Config.__dict__.update(parent_config.__dict__)
+
+
+def get_total_memory() -> int:
+    """Returns total amount of physical memory available in bytes"""
+    if platform.system() == "Windows":
+        from ctypes import Structure, c_int32, c_uint64, sizeof, byref, windll
+
+        # Use Win32 API : https://stackoverflow.com/questions/31546309/
+        class MemoryStatusEx(Structure):
+            _fields_ = [
+                ("length", c_int32),
+                ("memoryLoad", c_int32),
+                ("totalPhys", c_uint64),
+                ("availPhys", c_uint64),
+                ("totalPageFile", c_uint64),
+                ("availPageFile", c_uint64),
+                ("totalVirtual", c_uint64),
+                ("availVirtual", c_uint64),
+                ("availExtendedVirtual", c_uint64),
+            ]
+
+            def __init__(self):
+                self.length = sizeof(self)
+
+        win_mem = MemoryStatusEx()
+        if windll.kernel32.GlobalMemoryStatusEx(byref(win_mem)):
+            return int(win_mem.totalPhys)
+        else:
+            raise OSError
```

Comment:
deeply nested code is difficult to read – I would break this out into a function and have

```python
def get_total_memory():
    """...docstring..."""
    if platform.system() == "Windows":
        return _get_total_memory_on_windows()
    return os.sysconf("SC_PAGE_SIZE") * os.sysconf("SC_PHYS_PAGES")
```